### ISSUE TITLE
Portable edition: 'Unable to get youtube-dl extractor list'

### ISSUE THREAD

**Issue by awrog** [yafp/media-dupes#32]
Portable edition
Click button **Extractors**: 'Unable to get youtube-dl extractor list'
Downloading URL generates following error

![Screenshot - 12_30_2019 , 13_19_46](https://user-images.githubusercontent.com/12153667/71581775-242e4900-2b07-11ea-9b79-9bf178f40b5f.png)

Windows 10 Pro 1903

**Comment by yafp** [comment]
Hey,
first of all - thanks for the response.

seems like you got 2 issues right?

a) The "extractors" button does not show the list of supported youtube-dl extranctors
b) Downloading the url mentioned in your screenshot does not work.

Are you using media-dupes 0.3.0? Or an older version?

Pressing F12 opens the chrome developer tools - where you got access to a tab called 'Console'.
If you can reproduce the errors it might help if you could show screenshots of the console output of both errors.

Is the download function broken on your system for all urls or just this specific one?

**Comment by yafp** [comment]
URL for debugging

https://www.youtube.com/watch?v=RFsmqh-BD08

**Comment by yafp** [comment]
I just downloaded the portable version of 0.3.0 on my windows 10 (1903) machine.
Then i pressed the Extractors button.
And opened the dev console for more details.

Seems like i can't reproduce the issue you got on my windows machine.

![media-dupes_2019-12-30_20-41-06](https://user-images.githubusercontent.com/67605/71597870-2a431a80-2b45-11ea-8154-8413c018ae77.png)

**Comment by yafp** [comment]
Regarding your second issue: 
Tried downloading the youtube video you referenced.

I can reproduce that issue and i do know why it happens ;)
Error on my side. I will release a fix (version 0.4.0)
![BG1 - YouTube - Mozilla Firefox_2019-12-30_20-44-53](https://user-images.githubusercontent.com/67605/71598051-b35a5180-2b45-11ea-9556-161f107dd985.png)
 in the first or second week of january

**Comment by yafp** [comment]
@awrog 
regarding the second problem:
yes i can reproduce the error - but the main task still works.
The download is finished without issues on my problem - my download destination contains a 12 second movie clip

So summarize it:
yes 0.3.0 contains an error (see #30) where a variable is not defined - which results in the error message - which is misleading.
The download itself is working ;)

can you confirm that?

**Comment by awrog** [comment]
Thanks for your swift en thorough reply!
I 'm using the latest version (media-dupes-0.3.0-win-portable.exe, v0.3.0) Chrome: 78.0.3904.130

Console output in [attachment](https://github.com/yafp/media-dupes/files/4012419/console-output.txt).

I'm only using the portable version (for now).
Should I use the/an installer?

Thanks!

**Comment by yafp** [comment]
@awrog:
just to make it clear: the extractors button just shows a list of all supported services/domains.
In case you want to handle a for example youtube.com url - there is no need to press the extractors button after pasting the url.

In case the general workflow within media-dupes is unclear to you - consider using the "intro" button.

tldr:
* paste url
* press "Grab Audio" OR press "Grab video"

This does not explain why you are getting an error 9009 on the extractor function ;)


**Regarding:** 

> Should I use the/an installer?

In theory there should be function-wise no difference between the windows installer and the portable version.

**Comment by awrog** [comment]
fresh new attempt.
Also downloading does not work (see below) ![animation](https://user-images.githubusercontent.com/12153667/71666451-27bd0c80-2d61-11ea-961f-c73bdfcb84fe.gif)
Tried: media-dupes-0.3.0-win-x64.zip and media-dupes-0.3.0-win-ia32.zip

**Comment by yafp** [comment]
The upcoming version 0.4.0 will support in-app updates of the youtube-dl binary (the app media-dupes is using youtube-dl bundled to realize the downloads). This binary is heavily updated - to cover changes media pages are implementing.

This should help reacting fast to changes without having a need to update media-dupes itself with a new build.

It looks promising on my development system ...

**Comment by yafp** [comment]
dev note regarding error 9009: see sentry: MEDIA-DUPES-2D

**Comment by yafp** [comment]
@awrog 
seems like i found a majro bug in my build-workflow which might have caused the issues on windows. See #47
I'll release 0.4.0 in the next 24 hours i guess.

**Comment by yafp** [comment]
@awrog 
just released 0.4.2 which fixed another bug in the windows builds.
Pretty sure it should work now for you ;)

**Comment by awrog** [comment]
Works great!

**Comment by yafp** [comment]
@awrog 
thanks for the feedback.
Great to know it finally is working for you (and the other windows users) as well :)

Enjoy.
